Userland/Services/LaunchServer/Launcher.cpp

```diff
@@ -393,14 +393,16 @@ bool Launcher::open_file_url(const URL& url)
     Vector<DeprecatedString> additional_parameters;
     DeprecatedString filepath = url.serialize_path();
 
-    auto parameters = url.query().split('&');
-    for (auto const& parameter : parameters) {
-        auto pair = parameter.split('=');
-        if (pair.size() == 2 && pair[0] == "line_number") {
-            auto line = pair[1].to_int();
-            if (line.has_value())
-                // TextEditor uses file:line:col to open a file at a specific line number
-                filepath = DeprecatedString::formatted("{}:{}", filepath, line.value());
+    if (url.query().has_value()) {
+        auto parameters = url.query()->bytes_as_string_view().split_view('&');
```

Comment:
you could also call `for_each_split_view` here to further allocate less. (split_view returns a Vector :) )

Reply:
nice, wasn't aware of that function